site/guides/sponsoring-gas/gas-sponsorship-eligibility.md

```diff
@@ -0,0 +1,120 @@
+  - - meta
+    - property: og:description
+      content: Follow this guide to handle User Operations that are not eligible for gas sponsorship. Account Kit is a vertically integrated stack for building apps that support ERC-4337.
+  - - meta
+    - name: twitter:title
+      content: How to Handle User Operations Not Eligibile for Gas Sponsorship
+  - - meta
+    - name: twitter:description
+      content: Follow this guide to handle User Operations that are not eligible for gas sponsorship. Account Kit is a vertically integrated stack for building apps that support ERC-4337.
+---
+
+# How to Handle User Operations Not Eligibile for Gas Sponsorship
+
+As mentioned from the previous guide on [How to Sponsor Gas for a User Operation](./sponsoring-gas.md), with Account Kit dponsor gas fees for transactions via the [Gas Manager](https://docs.alchemy.com/docs/gas-manager-services).
+
+But what happens when the user operation you are sending fails to satisfy the criteria set in the gas manager policy? How do you check if the user operation is eligible for gas sponsorship before sending the user operation?
+
+If you do send the user operation that is not eligible for the gas sponsorship under your Gas Manager policy, `buildUserOperation` and `sendUserOperation` will fail due to the error thrown during the [`PaymasterMiddleware`](/packages/aa-core/provider/withPaymasterMiddleware.md) failure. You can follow the guide below to check for gas sponsorship eligibility in advance.
+
+## 1. How to Check for Gas Sponsorship Eligibility for a User Operation
+
+First, you can follow the same instructions from the previous guide on [How to Sponsor Gas for a User Operation](./sponsoring-gas.md) to set up your `SmartAccountProvider` and link your gas policy.
+
+::: code-group
+
+```ts [setup-alchemy-provider.ts]
+import { provider } from "./provider.ts";
+
+// Find your Gas Manager policy id at: // [!code focus:10]
+//dashboard.alchemy.com/gas-manager/policy/create
+const GAS_MANAGER_POLICY_ID = "YourGasManagerPolicyId";
+
+// Link the provider with the Gas Manager. This ensures user operations
+// sent with this provider get sponsorship from the Gas Manager.
+provider.withAlchemyGasManager({
+  policyId: GAS_MANAGER_POLICY_ID,
+});
+```
+
+<<< @/snippets/provider.ts
+
+:::
+
+Then, before you call `sendUserOperation` on the provider, you can use [`buildUserOperation`](/packages/aa-core/provider/buildUserOperation.md) to check if the user operation is eligible for gas sponsorship or not.
+
+Calling `buildUserOperation` method on the provider that is configured with the gas manager (`withAlchemyGasManager`) will run the middleware pipeline including the paymaster middleware, which will thrown an error if the user operation is not eligible for gas sponsorship under the policy.
+
+```ts [check-gas-sponsorship-eligibility.ts]
+let elligible = true;
+
+try {
+  await provider.buildUserOperation({
```

Comment:
looking at this, maybe I'm convinced it'd be a nice to have a convenience method for this >.>

something like

```ts
function checkSponsorshipEligibility(uo: ..., overrides: ...): boolean {
  return this.buildUserOperation(uo, overrides)
    .then(r => r.paymasterAndData !== "0x" && r.paymasterAndData != null)
    .catch(() => false);
}
```